### DISCUSSION TITLE
Write a guide for libary writers

### DISCUSSION THREAD

**Discussion by autoantwort** [microsoft/vcpkg#20739] [Q&A]
Libraries must support different thinks when they should be in vcpkg. @strega-nil you said a little bit about it in the cpp cast. 
1. They should prefer cmake.
2. They must support out of source dependencies.
3. There must be an explicit way to enable/disable dependencies when they are optional.
4. They should respect `BUILD_SHARED_LIBS`
5. `vcpkg_fixup_cmake_targets` should be necessary

Maybe also say something about other build systems. 
****

**Is your feature request related to a problem? Please describe.**
 nmwsharp/polyscope#47 asked what he should do so that the lib can be included in vcpkg, but couldn't find information.

**Proposed solution**
A guild with specific steps what library writers should/must do

**Comment by strega-nil** [comment]
@autoantwort This is one of those things that would be really nice, but isn't _really_ a vcpkg thing; it's more of a C++ community at large thing. I do wonder if we could contribute to it tho.

**Comment by autoantwort** [comment]
@strega-nil Yes, in general this is a c++ community thing. But I think there are some aspects that are special for vcpkg or more important than usual. And I think even for a general c++ library guideline (for buildsystems) vcpkg is a good place.

**Comment by JackBoosY** [comment]
In fact, we have a feature `create` to create the port draft.

**Comment by JackBoosY** [comment]
See https://github.com/microsoft/vcpkg/blob/master/docs/examples/packaging-github-repos.md.
We should improve our document.
